Create containers for testing into TravisCI

DrEdi opened the pull request:
Create script for run vagga container on TravisCI server #9

DrEdi committed:
Add requirement-test for testing

DrEdi committed:
Create test integrated with TravisCI

Add new container and command in vagga.yaml
Add commands in travis file for run vagga

DrEdi committed:
Change repo

webknjaz commented on .travis.yml:
```diff
@@ -1,5 +1,32 @@
+install: vagga _build test
+
+script:
+  - vagga test
```
You don't have to use sequence for a single command.

webknjaz commented on requirements-test.txt:
```diff
@@ -0,0 +1,2 @@
+pre-commit==0.10.1
+setuptools==34.0.2
```
Normal python installation usually contains setuptools, so this is only needed for vagga env. Please remove it from this file and add to vagga config

webknjaz commented on vagga.yaml:
```diff
@@ -51,6 +51,18 @@ containers:
+      - !Ubuntu xenial
+      - !PipConfig
+        dependencies: true
+      - !BuildDeps
```
You don't need to declare runtime `!Install` dependencies as `!BuildDeps` as they are available during build as well

webknjaz commented on vagga.yaml:
```diff
@@ -51,6 +51,18 @@ containers:
+        - git
+      - !Depends .pre-commit-config.yaml
+      - !Py3Requirements requirements-test.txt
+      - !Install
```
Please move this section up to `!BuildDeps` to avoid situation of installing python packages using `git+git://` URL when `git` is not installed yet.

webknjaz commented on vagga.yaml:
```diff
@@ -51,6 +51,18 @@ containers:
           postgresql: { mode: 0o777 }
+  test:
+    setup:
+      - !Ubuntu xenial
```
`Alpine` would work just fine and it will be built faster as it's lightweight.

DrEdi replied:
Nope. This code is properly works with `Ubunty xenial`, but it always fails with` Alpine`.

webknjaz replied:
Why?

DrEdi replied:
`[Errno 95] Not supported: '/tmp/pip-u800vb8q-build/testing/resources/working_symlink'")]`
End there're many many other with code 95

webknjaz replied:
And what does it mean? Any detail?

DrEdi replied:
'An unexpected error has occurred: CalledProcessError: Command: ('/root/.pre-commit/reponf39kcdj/py_env-default/bin/python3.5', '/root/.pre-commit/reponf39kcdj/py_env-default/bin/pip', 'install', '.')
Return code: 2
Expected return code: 0'
Huge list of [Errno 95]. Then command failed
I've noticed before that some commands do not run under all OS

webknjaz replied:
Can we improve/fix this?

DrEdi replied:
I tried some variants. This is not depends on version of alpine. First time I thought that maybe v3.4 of Alpine may be a bit deprecated - after I tried it with v3.5 - result was the same.I also installed it under Ubuntu trusty  and it works. 
I can be wrong but I think Ubuntu has some packages that does not have Alpine, that's why we have trouble with it...

webknjaz replied:
I'm gonna dig into it myself.
![](http://www.quickmeme.com/img/e7/e7633bedf897bb24ce668ac9c5df6bf88a58ff7e114d27606a756f4c4888a3f1.jpg)

DrEdi replied:
eeee, memes)

webknjaz replied:
Alright, it looks failing somewhere inside of `shutil.copytree()`'s system calls, which are file permission/type checks. I think we can leave ubuntu here.

webknjaz commented on vagga.yaml:
```diff
@@ -85,3 +97,7 @@ commands:
+  test: !Command
+    description: Run tests
+    container: test
+    run: pre-commit run --all-files
```
please expose separate `vagga pre-commit` command using `exec` syntax (`run` value is a sequeance)

DrEdi committed:
Fix remarks

Move setuptools install in vagga file
Delete BuildDeps from vagga
Move Install tag a bit above in vagga to avoide problems with git
Some code-style fix

webknjaz commented on vagga.yaml:
```diff
@@ -85,3 +95,7 @@ commands:
     children:
       app: *run-app
       db: *run-db
+  pre-commit: !Command
```
Where's `test` command?

webknjaz commented on .travis.yml:
```diff
@@ -1,5 +1,31 @@
+
+install: vagga _build test
+
+script: vagga pre-commit --all-files
```
Why did you change this?

DrEdi committed:
Revert changes and add new command to run pre-commit
